pds/server.go

```diff
@@ -356,6 +358,15 @@ func (s *Server) RunAPI(listen string) error {
 
 	e.HTTPErrorHandler = func(err error, ctx echo.Context) {
 		fmt.Printf("HANDLER ERROR: (%s) %s\n", ctx.Path(), err)
+
+		// TODO: need to properly figure out where http error codes for error
+		// types get decided. This spot is reasonable, but maybe a bit weird.
+		// reviewers, please advise
```

Comment:
I have been doing error codes in XRPC endpoint handlers like: https://github.com/bluesky-social/indigo/blob/main/labeler/xrpc_handlers.go#L317

We can catch and handle all possible errors that way, which actually feels alright in the golang err handling paradigm. I think adding error types for every possible error here would be laborious, and having it work more than one way would also be brittle.

I would catch here things that are really 5xx errors and give them better messages. Eg, database connection errors, or any upstream HTTP connection errors (for proxy requests or other instances we might need to do upstream requests, like PLC).

Reply:
I'm gonna do this better in a later refactor, this particular error gets returned a lot of places, and should very often be a 404 (but not always). Probably need some helper functions or something